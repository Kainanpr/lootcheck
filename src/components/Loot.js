/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../redux/actions/bitcoin';

export class Loot extends Component {
  static propTypes = {
    fetchBitcoin: PropTypes.func,
  };

  static defaultProps = {
    fetchBitcoin: () => {},
  };

  componentDidMount() {
    const { fetchBitcoin } = this.props;

    fetchBitcoin();
  }

  render() {
    return (
      <h3>Bitcoin balance:</h3>
    );
  }
}

const mapStateToProps = state => ({
  balance: state.balance,
  bitcoin: state.bitcoin,
});

const mapDispatchToProps = {
  fetchBitcoin,
};

export default connect(mapStateToProps, mapDispatchToProps)(Loot);
