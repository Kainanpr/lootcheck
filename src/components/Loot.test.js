import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Loot } from './Loot';

configure({ adapter: new Adapter() });

describe('Loot', () => {
  const mockFetchBitcoin = jest.fn();
  const props = { balance: 10, bitcoin: {}, fetchBitcoin: mockFetchBitcoin };
  const wrapper = mount(<Loot {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('when mounted', () => {
    it('dispatches the `fetchBitcoin()` method it receives from props', () => {
      expect(mockFetchBitcoin).toHaveBeenCalled();
    });
  });
});
