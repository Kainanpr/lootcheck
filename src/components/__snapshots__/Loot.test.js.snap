// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot renders properly 1`] = `
<Loot
  balance={10}
  bitcoin={Object {}}
  fetchBitcoin={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <h3>
    Bitcoin balance:
  </h3>
</Loot>
`;
